src/defaults/html.pipe.js

```diff
@@ -71,7 +70,7 @@ const htmlpipe = (cont, payload, action) => {
     .after(addHeaders)
     .after(tohtml) // end HTML post-processing
     .after(flag).expose('esi').when(esi) // flag ESI when there is ESI in the response
-    .error(selectStatus(production()));
+    .error(selectStatus);
```

Comment:
So selectStatus will print send nice error messages in a dev environment…

Reply:
Is there any particular motivation for that…

Reply:
> Not really sure why the pipeline step should'nt send nice messages in prod [was: dev]

Assuming you meant `prod`: error messages are a common attack vector: they divulge information about the system such as file names and paths, and can sometimes be crafted into XSS attacks.

Thinking about this, it comes clear that the current implementation is not ideal. Ideally, we should have a `detailedstatus` and `tersestatus` function that are called in the pipeline like this:

```
.error(chattystatus).unless(production())
.error(tersestatus)
```

This would make both status functions single-purpose and more easily reusable, based on the preferences of the pipeline authors (some which might agree with you that the convenience of chatty status messages outweighs the risk of security issues, even in production.

Reply:
My new code does perform checks whether we are in production; if we are in production no error message will be divulged, in dev an error message will be returned. This check has been moved into selectStatus itself; my assumption why it was previously given as a higher oder parameter is that it could be set for the purposes of testing; my update also supports testing with both cases; it uses mocks.
Since no error messages are divulged in prod, @trieloff can we resolve this?